projects/js-packages/publicize-components/src/components/media-section/index.js

```diff
@@ -40,34 +44,55 @@ const ADD_MEDIA_LABEL = __( 'Set Social Image', 'jetpack' );
 const REPLACE_MEDIA_LABEL = __( 'Replace Social Image', 'jetpack' );
 const REMOVE_MEDIA_LABEL = __( 'Remove Social Image', 'jetpack' );
 
-const ALLOWED_MEDIA_TYPES = [ 'image/jpeg', 'image/png' ];
-
 /**
  * Wrapper that handles media-related functionality.
  *
  * @returns {object} The media section.
  */
 export default function MediaSection() {
+	const [ validationError, setValidationError ] = useState( null );
 	const { attachedMedia, updateAttachedMedia } = useAttachedMedia();
+	const { connections } = useSocialMediaConnections();
+	const enabledConnections = connections.filter( connection => connection.enabled );
+
+	const { maxImageSize, allowedImageTypes, getValidationError } = useMediaRestrictions(
+		enabledConnections
+	);
 
 	const mediaObject = useSelect( select =>
 		select( 'core' ).getMedia( attachedMedia[ 0 ] || null, { context: 'view' } )
 	);
```

Comment:
I know this is not part of this change, but I think we may need to add `attachedMedia[ 0 ]` to the dependency array here.
```suggestion
	[ attachedMedia[ 0 ] ] );
```